microservice/src/test/java/de/gsi/microservice/concepts/majordomo/MajordomoBrokerTests.java

```diff
@@ -0,0 +1,262 @@
+        MajordomoBroker broker = new MajordomoBroker(1, BasicRbacRole.values());
+        // broker.setDaemon(true); // use this if running in another app that controls threads
+        final int openPort = Utils.findOpenPort();
+        broker.bind("tcp://*:" + openPort);
+        broker.start();
+
+        // using simple synchronous client
+        MajordomoClientV1 clientSession = new MajordomoClientV1("tcp://localhost:" + openPort, "customClientName");
+
+        {
+            final ZMsg replyWithoutRbac = clientSession.send("mmi.echo", DEFAULT_REQUEST_MESSAGE_BYTES); // w/o RBAC
+            assertNotNull(replyWithoutRbac, "reply message w/o RBAC token not being null");
+            assertNotNull(replyWithoutRbac.peekLast(), "user-data not being null");
+            assertArrayEquals(DEFAULT_REQUEST_MESSAGE_BYTES, replyWithoutRbac.pollLast().getData(), "MMI echo service request");
+        }
+
+        {
+            final ZMsg replyWithoutRbac = clientSession.send(DEFAULT_MMI_SERVICE, DEFAULT_MMI_SERVICE); // w/o RBAC
+            assertNotNull(replyWithoutRbac, "reply message w/o RBAC token not being null");
+            assertNotNull(replyWithoutRbac.peekLast(), "user-data not being null");
+            assertEquals("200", new String(replyWithoutRbac.pollLast().getData(), StandardCharsets.UTF_8), "known MMI service request");
+        }
+
+        {
+            final ZMsg replyWithoutRbac = clientSession.send(DEFAULT_MMI_SERVICE, DEFAULT_ECHO_SERVICE); // w/o RBAC
+            assertNotNull(replyWithoutRbac, "reply message w/o RBAC token not being null");
+            assertNotNull(replyWithoutRbac.peekLast(), "user-data not being null");
+            assertEquals("200", new String(replyWithoutRbac.pollLast().getData(), StandardCharsets.UTF_8), "known MMI service request");
+        }
+
+        {
+            // MMI service request: service should not exist
+            final ZMsg replyWithoutRbac = clientSession.send(DEFAULT_MMI_SERVICE, DEFAULT_REQUEST_MESSAGE_BYTES); // w/o RBAC
+            assertNotNull(replyWithoutRbac, "reply message w/o RBAC token not being null");
+            assertNotNull(replyWithoutRbac.peekLast(), "user-data not being null");
+            assertEquals("400", new String(replyWithoutRbac.pollLast().getData(), StandardCharsets.UTF_8), "unknown MMI service request");
+        }
+
+        {
+            // unknown service name
+            final ZMsg replyWithoutRbac = clientSession.send("unknownService".getBytes(StandardCharsets.UTF_8), DEFAULT_REQUEST_MESSAGE_BYTES); // w/o RBAC
+            assertNotNull(replyWithoutRbac, "reply message w/o RBAC token not being null");
+            assertNotNull(replyWithoutRbac.peekLast(), "user-data not being null");
+            assertEquals("501", new String(replyWithoutRbac.pollLast().getData()), "unknown service");
+        }
+
+        broker.stopBroker();
+    }
+
+    @Test
+    public void basicASynchronousRequestReplyTest() throws IOException {
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)